cookbooks/fb_users/spec/default_spec.rb

```diff
@@ -0,0 +1,87 @@
+# vim: syntax=ruby:expandtab:shiftwidth=2:softtabstop=2:tabstop=2
+#
+# Copyright (c) 2016-present, Facebook, Inc.
+# All rights reserved.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+# http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+#
+require './spec/spec_helper'
+require_relative '../libraries/default'
+
+recipe 'fb_users::default' do |_tc|
+  describe FB::Users do
+    let(:node) { Chef::Node.new }
+    context 'Validation' do
+      before(:each) do
+        node.default['fb_users'] = {
+          'user_defaults' => {},
+          'users' => {
+            'testuser' => {
+              'shell' => '/bin/zsh',
+              'gid' => 'testgroup',
+              'action' => :add,
+            },
+          },
+          'groups' => {
+            'testgroup' => {
+              'members' => ['testuser'],
+              'action' => :add,
+            },
+          },
+        }
+      end
+
+      it 'should not fail if we did not specify a protected uid range' do
+        stub_const('FB::Users::UID_MAP', { 'testuser' => { 'uid' => 42 } })
+        stub_const('FB::Users::GID_MAP', { 'testgroup' => { 'gid' => 4242 } })
+        FB::Users._validate(node)
```

Comment:
add an expectation here. `expect { FB::Users._validate(node) }.no raise_error(RuntimeError)`

Reply:
I would prefer to ensure it doesn't raise ANY error:

```
expect { FB::Users._validate(node) }.not_to raise_error
```

vs

```
expect { FB::Users._validate(node) }.not_to raise_error(RuntimeError)
```

Thoughts?

Reply:
The default error is Runtime, so I don't think it matters much, but yeah I'm fine with that.